### DISCUSSION TITLE
Error propagation in dependent queries

### DISCUSSION THREAD

**Discussion by viridia** [TanStack/query#2248] [Ideas]
The [documented example](https://react-query.tanstack.com/guides/dependent-queries) for dependent queries has a serious shortcoming, which is that errors are not propagated: if an error occurs in the upstream query, the error is simply dropped. Callers which use the output of the dependent query will never see the error, instead they will just wait forever, since the dependent query will never be enabled.

I would like to see a more comprehensive example that shows how to propagate errors to dependent queries.

**Comment by TkDodo** [comment]
I think it would just be something like:

```
 const userQuery = useQuery(['user', email], getUserByEmail)
 
 const userId = userQuery.data?.id
 
 // Then get the user's projects
 const projectsQuery = useQuery(
   ['projects', userId],
   getProjectsByUser,
   {
     // The query will not execute until the userId exists
     enabled: !!userId,
   }
 )

const error = projectsQuery.error ?? userQuery.error

```

**Comment by mercmobily** [comment]
I believe I am experiencing this
